code/move/solution/trymove.sol.cpp

```diff
@@ -1,39 +1,87 @@
-#include <iostream>
-#include <vector>
+#include <memory>
 #include <random>
-#include "NVector.sol.hpp"
-
-template<typename T>
-void swap(std::vector<T> &v, int a, int b) {
-    T mem = std::move(v[a]);
-    v[a] = std::move(v[b]);
-    v[b] = std::move(mem);
-    // even better
-    //swap(v[a], v[b]);
-}
+#include <vector>
 
-template<typename T>
-void randomize(std::vector<T> &v) {
-    // we randomize via len random inversions
-    std::size_t len = v.size();
-    std::default_random_engine e;
-    std::uniform_int_distribution<std::size_t> d{0, len};
-    for (std::size_t i = 0; i < len; i++) {
-        int a = d(e);
-        int b = d(e);
-        swap(v, a, b);
+/**
+ * An attempt to make a movable class. It allocates memory on construction, and has
+ * array semantics. (Prefer std::array or std::vector instead of hand-crafted solutions, though.)
+ * We made it copyable by providing copy constructor and assignment operator, but note that we
+ * are violating the rule of 5, so it's not moveable yet.
+*/
+class Moveable {
+    static constexpr std::size_t size = 10000;
+
+public:
+    Moveable() = default;
+    Moveable(Moveable const & other) :
+        Moveable()
+    {
```

Comment:
Why did you delegate to the default ctor?
```suggestion
    Moveable(Moveable const & other) :
    {
```

Reply:
Because initially, I used it to allocate. It's not needed, any more, but is there a reason not to delegate?

Reply:
Well, it's dead code, but otherwise there is no reason. Keep it if you want.